### ISSUE TITLE
Inverse-logit link produces implausible extrapolation — Piel replaced it for this reason

### ISSUE THREAD

**Issue by bschilder** [bschilder/genomeOS#103]
§7 specifies the standard binomial GLM link:

```
AC_i ~ Binomial(AN_i, expit(f(s_i) + β_design[i] + β_cohort[i]))
```

Fitting that to the real 332 MAP surveys produces a **sickle-cell hotspot across Finland and northern Scandinavia**, where the true allele frequency is approximately zero and there is not one observation within 800 km. Switching the reported summary to the median (#102) roughly halved it — from ~0.25 to ~0.10–0.14 — but did not remove it.

## This is a known failure of this exact model on this exact dataset

Piel et al.'s supplementary appendix describes fitting their 2010 inverse-logit model to this data:

> When we initially attempted to fit Piel et al.'s model to the current dataset and predict, we found that… the best fitting values result in **implausibly long right-hand tails** for the predictive distribution… strikingly high allele frequencies (**greater than 40%**) were predicted with small but practically significant probability (~0.1%). …the long right-hand tail **contained enough mass to skew all of the standard summary statistics**.
>
> To remedy this problem, we attempted several strategies including employing **Stukel's link function** in place of the more standard inverse logit, and modelling as a **skew-Gaussian process** rather than the standard Gaussian. The skew-Gaussian approach indicated that it could solve the problem, but eliminated the crucial conjugate relationship… **Ultimately, we employed an alternative flexible link function**… We were unable to infer its coefficients jointly with the other model parameters in a fully Bayesian manner due to poor MCMC mixing, so we adopted an **empirical fitting approach**.

So the reference implementation we are scored against tried the model §7 specifies, found it inadequate **for this dataset**, and replaced the link. §7 should not specify a link that its own validation target had to abandon.

## Why it matters beyond aesthetics

The mask does not catch it. Those Nordic cells classify as `interpolated`, because the fitted correlation range is 1,410 km and European surveys fall inside 2ρ. **The data-support mask answers "is there data nearby?", never "is this value credible?"** — so a link-function artifact lands inside supported territory and renders as confident signal. §4's central defence against manufactured clines does not cover this failure mode.

## Options

1. **Flexible link**, as Piel used — closest to the reference, and their empirical-Bayes fitting recipe is documented in the appendix.
2. **Stukel's link** — they tried it; the appendix does not say it failed, only that it was among the attempts.
3. **Constrain extrapolation variance instead of the link.** The fitted GP amplitude was 3.32 on the logit scale, which is what lets the field wander far from data. Tightening that prior is cheap but treats the symptom.
4. **Beta-binomial** (#83) addresses overdispersion, which is related but not the same defect.

Also fixed alongside this, and worth noting as a separate contributor: the GP now lives on the **unit sphere** rather than on separately-standardised (lon, lat), which had made the kernel anisotropic in a way driven by survey placement rather than geography.

`wants-expert-review`, `priority:critical` — this gates #45, since golden test 1 cannot pass while the model invents Nordic sickle cell.

Related: #34, #35, #102, #83, #45.

**Comment by bschilder** [comment]
**Closing as completed. The artifact is gone, and — more importantly — the mask now catches the territory it was rendering into.**

## The specific defect, re-measured

This issue was filed against a sickle-cell hotspot of **0.10–0.14 across Finland and northern Scandinavia**, where the true frequency is ~0 and no observation lies within 800 km. Measured on the current fit (857 surveys, inducing-point GP on the unit sphere, beta-binomial likelihood):

| region | posterior median | max |
|---|---|---|
| **Finland** (60–70N, 20–31E) | **0.0103** | 0.0105 |
| Nordic (55–71N, 4–31E) | 0.0104 | 0.0118 |
| Europe (45–71N, −10–40E) | 0.0106 | 0.0176 |
| West Africa (4–16N, −17–15E) | 0.0757 | 0.1412 |

Roughly a **ten-fold reduction** in the Nordic values, and West Africa now sits ~7× above Europe. The global maximum on land is **0.162 at 8.4S 17.5E** — the Congo Basin, which is where it belongs.

## The part that matters more

This issue's sharpest observation was not the number, it was that **the mask could not see the problem**:

> Those Nordic cells classify as `interpolated`, because the fitted correlation range is 1,410 km and European surveys fall inside 2ρ. The data-support mask answers "is there data nearby?", never "is this value credible?"

That is no longer true. The fitted correlation range fell from **1,410 km to 604 km**, so 2ρ fell from 2,820 km to **1,208 km**, while Finland's nearest survey is **1,345–2,293 km** away. Result:

```
Finland: 52 cells,  support = {'unknown': 52},  cells actually rendered = 0
Nordic : 159 cells, support = {'interpolated': 63, 'unknown': 96}, max rendered = 0.0118
```

**Finland is now outside supported territory and is not rendered at all.** The 63 Nordic cells that do render sit 449–1,208 km from real surveys in Denmark and the Baltic, so rendering them is legitimate, and they max out at 1.2%.

## How it was fixed — not by changing the link

The link is still `expit`. The resolution came through this issue's **option 3, "constrain extrapolation variance instead of the link"**, reached as a consequence of three other changes rather than as a direct attack:

- **Beta-binomial likelihood** (#83) — under the binomial, the only way to absorb survey-to-survey variance was to inflate the spatial field; GP amplitude fell from 11.5 (non-converged) to 1.79. A field that wanders less far from data extrapolates less wildly.
- **The GP moved to the unit sphere**, so the kernel is isotropic in real geography rather than in separately-standardised (lon, lat). This issue already noted this as a contributor.
- **Inducing-point conditioning** (#108) — spacing matched to the correlation range, which is what let ρ be estimated as 604 km rather than absorbed into a degenerate fit.

## What is deliberately not claimed

`expit` remains, so the *mechanism* that produced long right-hand tails still exists in principle. Piel's appendix hit it on a comparable dataset, and a future variant with sparser data could surface it again. Two guards now stand where none did: ρ is small enough that unsupported regions fall outside 2ρ and mask out, and the fit is refused outright if it does not converge (§12).

**If a hotspot ever appears in unsupported territory again, reopen this rather than treating it as new** — the flexible-link and Stukel options above remain the documented next moves.

**Comment by bschilder** [comment]
Reopening is warranted because the completed 994-survey burden run exposes the same mechanism at a scale the earlier visual check could not see. The old Nordic hotspot is mostly masked, but the inverse-logit surface still leaves a supported ~1% background that is unacceptable after population-weighted aggregation.

## New evidence

The canonical fit in PR #301 converged cleanly (max R-hat 1.0059, min ESS 767.6, zero divergences) and learned a 715 km range. Nevertheless:

| Country | Direct survey evidence in the 994-row subset | Supported-cell population-weighted median AF | Model HbAS births | Piel HbAS births |
|---|---:|---:|---:|---:|
| China | AC=0 / AN=2,000 | 0.57% | 185,244 | 43 |
| Indonesia | no retained survey | 0.84% | 64,602 | 328 |
| Afghanistan | AC=0 / AN=234 | 0.81% | 19,761 | 144 |

China has 4,969 supported H3 res-4 cells (5 `observed`, 4,964 `interpolated`) and only 400 `unknown`; Indonesia has 785 supported cells. This is therefore not fixed by the current mask. Finland still has 25 supported fringe cells at a median 0.975%, although 223/248 cells are correctly `unknown`.

The fitted beta-binomial concentration is ~46. At p=0.78%, AC=0 of AN=2,000 has about 25% probability under that observation model, versus 1.6e-7 under a binomial. This explains why a large zero survey exerts little downward pressure. Replacing beta-binomial with binomial is not the answer: #83 already established that the binomial HbS fit fails to mix (R-hat 2.61, ESS 2, amplitude 11.5).

The 2010-aligned golden result fails in opposite directions: HbAS is high globally at 6.471M versus 5.476M, while HbSS is low at 253k versus 312k. The broad country ranking is learned (Spearman 0.826 HbAS, 0.927 HbSS), so a global multiplier would destroy useful structure without fixing both tails.

## Bounded next comparison

Compare the current inverse-logit model with one predeclared flexible-link arm derived from the alternatives already documented in this issue. Keep the 994 observations, cohort/design terms, beta-binomial likelihood, inducing geometry, seed, posterior budget, support rule, and 2010 denominator fixed.

Choose between arms using dependency-aware spatial held-out **count** prediction under the global modeling plan's promotion defaults; fit any link parameters only inside each training fold. Piel national estimates are then a golden integration check, not a tuning target. Report convergence, large-zero posterior predictive probabilities, non-endemic background, endemic peaks, HbAS/HbSS parity, and every failed fold. Reject an arm that improves reused Piel totals while worsening held-out count score or calibration.

Do not add malaria, EVI, friction, or other MAP/environmental rasters in this comparison. They are separate covariate hypotheses and receive no priority from their source or topic. A hurdle/nonstationary model remains a later alternative only if the narrower link experiment fails.

Expert review should focus on whether a flexible link addresses the background floor without misusing the beta-binomial dispersion, and on faithful implementation of Piel's documented parameterization.

Full run identity, figure, and metrics: PR #301; #45 comment 5690651439.

**Comment by bschilder** [comment]
Primary-method check completed against the open Piel et al. supplementary appendix (PMC3547249, `mmc1.pdf`, Web Appendix 1 pp. 12–13).

The selected link is exactly:

```
h(x) = sum(c_i * x**i, i=0..3)
g(x) = inverse_logit(h(x))
```

Piel estimated `c` outside the spatial MCMC, constrained `h` to be monotone/invertible, and then treated the coefficients as fixed. Their procedure fits a nonspatial normal model to transformed survey frequencies, plots its posterior-predictive CDF against the empirical CDF, and fits the cubic by constrained least squares.

Two reproducibility limits must be explicit before implementation:

1. The fitted coefficient values are not published in the appendix, so we cannot claim to reproduce their link by copying constants.
2. The appendix prints the smoothed survey value as `(k_i + 1) / (2*n_i + k_i + 2)`. For `k_i ~ Binomial(2*n_i, p_i)` with a uniform prior, the standard conjugate posterior mean is `(k_i + 1) / (2*n_i + 2)`. The printed expression and the stated derivation disagree. We must not silently choose one.

Therefore the first reviewable artifact should be a pure coefficient-fitting preflight that labels and compares:

- `piel_printed_2013`: the equation exactly as printed;
- `uniform_binomial_conjugate`: the stated model's standard posterior mean.

It should report monotonicity domain, fitted coefficients, input fingerprints, transformed-frequency distribution, large-zero behavior, and sensitivity. Neither result enters the surface fitter until this ambiguity is reviewed. If the printed formula is confirmed as a typo from another primary source or author code, record that evidence; otherwise retain both as explicit experimental arms.

The appendix uses the HbS survey observations only. It provides no reason to add malaria, EVI, friction, or any other environmental raster to this link experiment.

Primary sources:

- article record/full text: https://europepmc.org/articles/PMC3547249
- supplementary bundle API: https://www.ebi.ac.uk/europepmc/webservices/rest/PMC3547249/supplementaryFiles

**Comment by bschilder** [comment]
Preflight evidence is now in PR #305 (commit `2055580ba66284f94d16d36502c3e72db98f6c98`).

Using the 994-row Piel-comparable count artifact, 963 observations pass the appendix's `AN >= 50` rule. I reconstructed both explicit readings of the ambiguous smoothing equation with named Hazen-rank / plug-in-normal assumptions and a globally monotone cubic:

- printed equation `(AC + 1) / (AN + AC + 2)`: logit RMSE 0.19424;
- uniform-binomial conjugate `(AC + 1) / (AN + 2)`: logit RMSE 0.19459.

At the canonical spatial fit's posterior-median intercept (`x = -4.599069`), inverse-logit gives 0.9961%, while the two reconstructed links give 1.2126% and 1.1939%. They increase the diffuse background by 21.7% and 19.9%, and their major effect is upper-tail compression. That does not target the campaign's high-background/weak-peak failure mode.

Decision: do not spend a GPU spatial fit on this link now. This is a bounded negative preflight rather than a proof about every joint refit; revisit if the original coefficients/full procedure surface, or inside a candidate that separately models low background and localized peaks. Any later candidate must beat the unchanged inverse-logit baseline on held-out counts, calibration, supported-only burden, and peak/background contrasts.

Source boundary: this uses count-bearing HbS survey observations because they instantiate the Piel benchmark. MAP malaria-risk, EVI, friction, and other modeled environmental layers were not used and receive no special priority. They require a named variant-specific mechanism, dependency/source audit, unchanged holdouts, and controlled incremental comparison under WP3.

The PR contains the aggregate receipt, curves, quantiles, and review figure. It records no spatial fit, no publication eligibility, and no environmental covariates. Full local CI passed with PyTensor's compile cache placed in writable `/private/tmp`.

**Comment by bschilder** [comment]
The next bounded alternative is complete in PR #306 (commit `b27540dba1f640a2365b31c3ffe2f7f06333e215`), and it also fails before a spatial fit.

## Scientific question

Does an explicit absent component improve held-out count prediction after the beta-binomial already represents ordinary sampling zeros and survey overdispersion? An observed zero is never labelled as structural absence: at AC=0 the candidate adds extra-zero mass to the ordinary beta-binomial zero mass.

The same `AN >= 50` subset has 963 observations from 384 cohorts: 200 zero and 763 positive counts. Five folds hold out whole cohorts; split cohorts are a hard error.

## Held-out result

Zero-inflated minus ordinary beta-binomial normalized log score:

- cohort-macro mean: **−0.001328 nats/observation**;
- row-weighted mean: **−0.007840**;
- zero-count rows: **+0.074161**;
- positive-count rows: **−0.029335**;
- folds improved: 4/5, with the remaining fold at **−0.09309** cohort-macro.

Every optimizer start succeeded. The full-data likelihood has a clear optimum near 7.9% extra-zero mass and improves in-sample log likelihood by 6.74, but it raises the conditional mean from 4.36% to 4.75%. Its marginal mean is 4.372%, slightly above the ordinary beta-binomial's 4.356%. Thus it reallocates probability from positive counts to zeros without lowering the aggregate mean.

## Decision

Do not build or run a spatial zero-inflated beta-binomial from this candidate. It fails the predeclared cohort-macro and no-positive-regression gates. This rejects the global probability-mass-transfer mechanism, not every nonstationary model; any later candidate must target low background and localized peaks directly and pass unchanged count gates before burden parity is consulted.

No MAP environmental product was used or promoted. The PR contains aggregate fold metrics, likelihood profile, hashes, receipt, review figure, pure tests, and a full passing local suite.

**Comment by bschilder** [comment]
Stukel's link has now been added to the same bounded preflight in PR #305 (commit `32d43734b33395c59450ad8515b03037cb982e98`). It reaches the same negative decision under both preserved smoothing equations.

The comparison uses affine empirical-quantile alignment because the Stukel fit has its own location and scale. Applying both links at the same raw latent number would compare different quantiles. Every fitted HbS Stukel latent is negative, so only the negative-tail shape is identified; the positive-tail shape is fixed at the inverse-logit value zero and reported as unfitted.

Results at the canonical inverse-logit background of 0.9961%:

- printed smoothing equation: aligned Stukel 1.2046%, empirical-logit RMSE 0.21247;
- conjugate smoothing equation: aligned Stukel 1.1878%, empirical-logit RMSE 0.21033;
- the corresponding cubic RMSEs are lower at 0.19424 and 0.19459;
- a 10% inverse-logit operating point becomes 9.0867% and 9.3765% under Stukel.

Both arms therefore raise diffuse background and compress peaks. Do not spend a GPU spatial fit on Stukel now. This rejects the fitted one-sided Stukel mechanism for the current failure; it does not claim that an unobserved positive tail or every joint spatial refit is impossible.

The pure implementation matches published Stukel reference values, all five optimizer starts converge, aggregate artifacts replay byte-for-byte, the full local suite passes, and GitHub CI is running. No environmental covariate was used. MAP survey observations remain one count source for the Piel benchmark; MAP environmental products have no privileged status and are not implied next steps.

**Comment by bschilder** [comment]
PR #322 implements the source-neutral B1 local count comparator from #307 and tests it on the byte-identical 994-row HbS B0/B2 benchmark.

The prespecified result rejects the hard local smoother as a global solution: 500/1,000/2,000 km candidates, a 300 km leakage buffer, at least two local observations, at least 100 weighted alleles, and a 50% training-only inner emission gate leave all five outer folds infeasible (0/994 emitted). A separately labeled post-hoc fixed-2,000 km sensitivity emits 218/994 rows. On that support-limited subset, positive-count rows improve by +148.03 nats/observation versus matched B0, but zero rows regress by −7.40; 95% coverage remains 37.1%, and macro MAE is slightly worse.

This narrows the next direction for #103: there is useful local endemic signal, but widening a stationary local kernel would trade honest refusal for renewed global smearing. A nonstationary or gated spatial model should preserve ordinary count zeros, shrink non-endemic background, retain explicit unsupported states, and pass zero/positive gates separately. MAP has no special status in this result; only the frozen count-bearing observations enter, with no environmental or publisher feature.

Evidence note and review figure: https://github.com/bschilder/genomeOS/pull/322
